kratos/solving_strategies/strategies/butcher_tableau.h

```diff
@@ -120,12 +122,19 @@ class ButcherTableau
      *          intended to be used with std::inner_product.
      */
 
-    ArraySlice GetMatrixRow(const unsigned int SubStepIndex) const
+    std::tuple<RowType::const_iterator, RowType::const_iterator> GetMatrixRow(const unsigned int SubStepIndex) const
     {
-        return ArraySlice{
-            mA[SubStepIndex - 1].begin(),
-            mA[SubStepIndex - 1].begin() + SubStepIndex
-        };
+        return {mA[SubStepIndex - 1].begin(), mA[SubStepIndex - 1].begin() + SubStepIndex};
```

Comment:
```suggestion
        return {GetMatrixRowBegin(SubStepIndex), GetMatrixRowEnd(GetMatrixRowEnd)};
```

This avoids duplicate code, should compile to the same code.

Reply:
Makes sense :+1: .